pkg_resources/tests/test_pkg_resources.py

```diff
@@ -190,6 +191,83 @@ def test_setuptools_not_imported(self):
         subprocess.check_call(cmd)
 
 
+def touch_file(path):
+    """
+    Create an empty file.
+    """
+    with open(path, 'w') as f:
+        f.write('')
+
+
+def make_distribution_no_version(tmpdir, basename):
+    """
+    Create a distribution directory with no file containing the version.
+    """
+    # Convert the LocalPath object to a string before joining.
+    dist_dir = os.path.join(str(tmpdir), basename)
+    os.mkdir(dist_dir)
+    # Make the directory non-empty so distributions_from_metadata()
+    # will detect it and yield it.
+    touch_file(os.path.join(dist_dir, 'temp.txt'))
+
+    dists = list(pkg_resources.distributions_from_metadata(dist_dir))
+    assert len(dists) == 1
+    dist, = dists
+
+    return dist, dist_dir
+
+
+def check_version_missing(dist, metadata_path, expected_text,
+    expected_dist_type):
+    """
+    Check accessing dist.version when the "Version" header is missing.
+    """
+    with pytest.raises(ValueError) as excinfo:
+        dist.version
+
+    err = str(excinfo)
+    # Include a string expression after the assert so the full strings
+    # will be visible for inspection on failure.
+    assert expected_text in err, str((expected_text, err))
+
+    # Also check the args passed to the ValueError.
+    msg, dist = excinfo.value.args
+    assert expected_text in msg
+    # Check that the message portion contains the path.
+    assert metadata_path in msg, str((metadata_path, msg))
+    assert type(dist) == expected_dist_type
+
+
+class TestDistribution:
+
+    def test_version_missing(self, tmpdir):
```

Comment:
I think it would be best to refactor `TestDistribution.test_version_missing` and `TestDistributionInfo.test_version_missing` as a single parametrized test that takes `expected_dist_type` as a parameter.

That would also allow you to move `check_version_info` inline into the parametrized test itself.